nibabel/bv.py

```diff
@@ -0,0 +1,703 @@
+                unpack(hdr_dict[key]['dt'],
+                       fileobj.read(calcsize(hdr_dict[key]['dt'])))[0]
+    return hdr_dict
+
+
+def pack_BV_header(hdr_dict):
+    """Pack the header of a BV file format for (binary) writing to a file.
+
+    This function can be (and is) called recursively to iterate through nested
+    fields (e.g. the prts field of the VTC header).
+
+    Parameters
+    ----------
+    hdrDict: OrderedDict
+       hdrDict that contains the fields and values to for the respective
+       BV file format.
+       A prototype for this OrderedDict is generated during init() for each
+       file type (e.g. see _make_vtc_hdrDict() in bv_vtc.py) and maintains all
+       the fields in the correct order to write a valid header when saving.
+    """
+    binaryblock = ''
+    for key in hdr_dict:
+        # handle zero-terminated strings
+        if hdr_dict[key]['dt'] == '<s':
+            binaryblock += pack('<' + str(len(hdr_dict[key]['value'])+1) + 's',
+                                hdr_dict[key]['value'] + '\x00')
+        # handle array fields
+        elif hdr_dict[key]['dt'] == 'multi':
+            # check the length of the array to expect
+            for i in range(hdr_dict[hdr_dict[key]['nField']]['value']):
+                # recusively iterate through the fields of all items
+                # in the array
+                binaryblock += pack_BV_header(hdr_dict[key]['value'][i])
+        else:
+            binaryblock += pack(hdr_dict[key]['dt'], hdr_dict[key]['value'])
+    return binaryblock
+
+
+def calc_BV_header_size(hdr_dict):
+    """Calculate the binary size of a hdrDict for a BV file format header.
+
+    This function can be (and is) called recursively to iterate through nested
+    fields (e.g. the prts field of the VTC header).
+
+    Parameters
+    ----------
+    hdrDict: OrderedDict
+       hdrDict that contains the fields and values to for the respective
+       BV file format.
+       A prototype for this OrderedDict is generated during init() for each
+       file type (e.g. see _make_vtc_hdrDict() in bv_vtc.py) and maintains all
+       the fields in the correct order to write a valid header when saving.
+    """
+    hdr_size = 0
+    for key in hdr_dict:
+        # handle zero-terminated strings
+        if hdr_dict[key]['dt'] == '<s':
+            hdr_size += calcsize('<' +
+                                 str(len(hdr_dict[key]['value'])+1) + 's')
+        # handle array fields
+        elif hdr_dict[key]['dt'] == 'multi':
+            # check the length of the array to expect
+            for i in range(hdr_dict[hdr_dict[key]['nField']]['value']):
+                # recusively iterate through the fields of all items
+                # in the array
+                hdr_size += calc_BV_header_size(hdr_dict[key]['value'][i])
+        else:
+            hdr_size += calcsize(hdr_dict[key]['dt'])
+    return hdr_size
+
+
+class BvError(Exception):
+
```

Comment:
OK, I removed the blank lines and switched from pep8 to pylint (which accepts these blank lines).